src/system/PageLockModule/Entity/Repository/PageLockRepository.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * This file is part of the Zikula package.
+ *
+ * Copyright Zikula Foundation - http://zikula.org/
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zikula\PageLockModule\Entity\Repository;
+
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\QueryBuilder;
+use InvalidArgumentException;
+
+/**
+ * Repository class used to implement own convenience methods for performing certain DQL queries.
+ *
+ * This is the repository class for pagelock entities.
+ */
+class PageLockRepository extends EntityRepository
+{
+    /**
+     * Returns amount of active locks.
+     *
+     * @param string $lockName  Name of lock.
+     * @param string $sessionId Identifier of current session.
+     *
+     * @return integer
+     *
+     * @throws InvalidArgumentException Thrown if invalid parameters are received
+     */
+    public function getActiveLockAmount($lockName, $sessionId)
+    {
+        // check parameters
+        if ($lockName == '' || $sessionId == '') {
+            throw new \InvalidArgumentException('Invalid parameter received.');
+        }
+
+        $qb = $this->getEntityManager()->createQueryBuilder();
```

Comment:
in a repo class, you can do:

`$qb = $this->createQueryBuilder('tbl')
    ->select('count(tbl.id)');

the `from` is not required.